File: BRaVe-Mobile-Backend/Models/data_payload/Individual.cs
using System.Text.Json.Serialization;

namespace BRaVe_Mobile_Backend.Models.data_payload
{
    public class Individual
    {
        [JsonPropertyName("uuid")]
        public Guid Uuid { get; set; }    
        // UNIQUEIDENTIFIER PRIMARY KEY
        [JsonPropertyName("householdId")]
        public string HouseholdId { get; set; }  
        // UNIQUEIDENTIFIER NOT NULL
        [JsonPropertyName("individualId")]
        public int IndividualId { get; set; }             // INT NOT NULL

        [JsonPropertyName("firstName")]
        public string FirstName { get; set; }             // NVARCHAR(255) NULL

        [JsonPropertyName("middleName")]
        public string MiddleName { get; set; }            // NVARCHAR(255) NULL

        [JsonPropertyName("lastName")]
        public string LastName { get; set; }              // NVARCHAR(255) NULL

        [JsonPropertyName("dob")]
        public object Dob { get; set; }                // DATE NULL

        [JsonPropertyName("ageInYears")]
        public object AgeInYears { get; set; }              // INT NULL

        [JsonPropertyName("ageInMonths")]
        public object AgeInMonths { get; set; }             // INT NULL

        [JsonPropertyName("ageInDays")]
        public object AgeInDays { get; set; }               // INT NULL

        [JsonPropertyName("relationship")]
        public int Relationship { get; set; }            // INT NULL

        [JsonPropertyName("gender")]
        public int Gender { get; set; }                  // INT NULL
                                                         //public string DataPoints { get; set; }            // NVARCHAR(MAX)
        [JsonPropertyName("dpAnswers")]
        public Dictionary<int, string> dataPoints { get; set; }                 // NVARCHAR(MAX)

        public string ExternalIndividualId { get; set; }  // NVARCHAR(100) NULL

        [JsonPropertyName("photoBase64")]
        public string PhotoBase64 { get; set; }           // NVARCHAR(MAX) NULL

        [JsonPropertyName("biometricBase64")]
        public string BiometricBase64 { get; set; }           // NVARCHAR(MAX) NULL

        [JsonPropertyName("biometricNotCollected")]
        public BiometricNotCollected biometricNotCollected { get; set; }

        public int tenantId { get; set; }                 // INT NOT NULL

        [JsonPropertyName("createdBy")]
        public string createdByUserId { get; set; }            // VARCHAR(50) NOT NULL

        [JsonPropertyName("createdOnMs")]
        public long createdOnMs { get; set; }                // DATETIME NOT NULL

        [JsonPropertyName("updatedBy")]
        public string updatedByUserId { get; set; }            // VARCHAR(50)

        [JsonPropertyName("updatedOnMs")]
        public long updatedOnMs { get; set; }               // DATETIME NULL
    }

}

File: BRaVeMobile/app/src/main/java/intl/iom/bravemobile/models/registrations/Individual.java
package intl.iom.bravemobile.models.registrations;

import android.os.Build;

import java.util.Date;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

import intl.iom.bravemobile.models.activities.ConsentsFeedback;

public class Individual
{
    public int individualId;
    public UUID uuid = UUID.randomUUID();
    public String firstName;
    public String middleName;
    public String lastName;
    public Date dob;
    public Integer ageInYears;
    public Integer ageInMonths;
    public Integer ageInDays;
    public Integer relationship;
    public Integer gender;
    public String photoBase64;

    public BiometricNotCollected biometricNotCollected;
    public String biometricBase64;
    public Map<Integer,String> dpAnswers;

    public int  deleted;
    public Date updated_on;

    public String householdId;
    public ConsentsFeedback feedback;
    public String createdBy;
    public long createdOnMs;
    public String updatedBy;
    public long updatedOnMs;


    public db_staging_individual getStagingData()
    {
        return new db_staging_individual(householdId,individualId,uuid,firstName,middleName,lastName,dob,ageInYears,ageInMonths,ageInDays,relationship,gender,photoBase64, biometricBase64, biometricNotCollected ,dpAnswers,feedback, createdBy, updatedBy, createdOnMs, updatedOnMs);
    }

    public boolean isBiometricCollected()
    {
        return biometricBase64!=null && !biometricBase64.isEmpty();
    }

    public boolean isPhotoCollected()
    {
        return photoBase64!=null && !photoBase64.isEmpty();
    }


    public static Individual getEmpty()
    {
        Individual ind = new Individual();
        ind.firstName = "-";
        ind.middleName ="-";
        ind.lastName = "-";
        ind.relationship = -1;
        ind.dob = null;
        ind.ageInYears = null;
        ind.ageInMonths = null;
        ind.ageInDays = null;
        ind.gender = -1;
        ind.photoBase64 = null;
        ind.biometricBase64 = null;
        return ind;
    }

}